core/common/src/test/kotlin/app/k9mail/core/common/outcome/OutcomeUnwrapExtensionsKtTest.kt

```diff
@@ -0,0 +1,56 @@
+package app.k9mail.core.common.outcome
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import org.junit.Test
+
+class OutcomeUnwrapExtensionsKtTest {
+
+    @Test
+    fun `should unwrap value of success outcome`() {
+        val outcome = Success("success") as Outcome<String, String>
+
+        val unwrappedValue = outcome.unwrapValue()
+
+        assertThat(unwrappedValue).isEqualTo("success")
+    }
+
+    @Test(expected = IllegalStateException::class)
+    fun `should throw error when unwrapping value of failure outcome`() {
+        val outcome = Failure("failure") as Outcome<String, String>
+
+        outcome.unwrapValue()
+    }
```

Comment:
By changing the throwing tests to the following structure, we can get rid of the type casts.

```suggestion
    @Test
    fun `should throw error when unwrapping value of failure outcome`() {
        val outcome = Failure("failure")

        assertFailure {
            outcome.unwrapValue()
        }.isInstanceOf<IllegalStateException>()
    }
```